postprocessing/visualization/tools/extractDataFromUnstructuredOutput.py

```diff
@@ -15,23 +15,78 @@
 parser.add_argument("--backend", type=str, choices=["hdf5", "raw"], default="hdf5", help="backend used: raw (.bin file), hdf5 (.h5)")
 parser.add_argument("--last", dest="last", default=False, action="store_true", help="output last time step")
 parser.add_argument("--idt", nargs="+", help="list of time step to write (ex $(seq 7 3 28))", type=int)
+parser.add_argument(
+    "--xfilter",
+    nargs=2,
+    metavar=("xmin", "xmax"),
+    help="output only cells with x center coordinate in range xmin xmax",
+    type=float,
+)
+parser.add_argument(
+    "--yfilter",
+    nargs=2,
+    metavar=("ymin", "ymax"),
+    help="output only cells with y center coordinate in range ymin ymax",
+    type=float,
+)
+parser.add_argument(
+    "--zfilter",
+    nargs=2,
+    metavar=("zmin", "zmax"),
+    help="output only cells with z center coordinate in range zmin zmax",
+    type=float,
+)
+parser.add_argument("--no_ocean", dest="no_ocean", default=False, action="store_true", help="remove sea-surface cells")
+
 args = parser.parse_args()
 
 sx = seissolxdmf.seissolxdmf(args.xdmfFilename)
+xyz = sx.ReadGeometry()
 connect = sx.ReadConnect()
-nElements = sx.nElements
+
+xyzc = (xyz[connect[:, 0], :] + xyz[connect[:, 1], :] + xyz[connect[:, 2], :]) / 3.0
+
+
+def filter_cells(coords, filter_range):
+    m = 0.5 * (filter_range[0] + filter_range[1])
+    d = 0.5 * (filter_range[1] - filter_range[0])
+    return np.where(np.abs(coords[:] - m) < d)[0]
+
+
+ids = range(0, sx.nElements)
+if args.xfilter:
+    id0 = filter_cells(xyzc[:, 0], args.xfilter)
+    ids = np.intersect1d(ids, id0) if len(ids) else id0
+if args.yfilter:
+    id0 = filter_cells(xyzc[:, 1], args.yfilter)
+    ids = np.intersect1d(ids, id0) if len(ids) else id0
+if args.zfilter:
+    id0 = filter_cells(xyzc[:, 2], args.zfilter)
+    ids = np.intersect1d(ids, id0) if len(ids) else id0
+if args.no_ocean:
+    id0 = np.where(np.abs(xyzc[:, 2]) > 1e-3)[0]
+    ids = np.intersect1d(ids, id0) if len(ids) else id0
```

Comment:
This filter removes all cells with the z component of their respective center >0, right? Is this really the intended use here?

Reply:
Yes, I also fear it is very specific. What if someone prefers enu  instead of ned orientation? Also you would remove parts, where the topography is above sea level. Since you can achieve this behavior with the z-filter anyways, I think we could remove the ocean filter.

Reply:
I've removed the option.